src/dnsvizor_web_interface.md

```diff
@@ -0,0 +1,79 @@
+# Web Interface & API
+
+DNSvizor exposes a web-based management interface and a DNS-over-HTTPS (DoH) resolver.
+
+## Authentication
+Access to sensitive endpoints (like the blocklist or configuration) requires authentication.
+
+- **Password Setup:** You must pass the `--password` argument at startup. If omitted, endpoints requiring authentication will not be accessible.
+- **Method:** The interface uses [HTTP Basic Auth](https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/Authentication).
+
+![Dnsvizor web authentication](../images/auth.png)
+
+Any username will work, but the password must the be same password you used to startup the unikernel.
+
+## Dashboard Endpoints
+
+The web server listens on the configured HTTPS port (default 443).
+
+- `/` or `/dashboard` 
+    - **method**: `GET` 
+    - **description**: Displays statistics such as query rates etc. 
+    - **authenticated**: false
+
+![Dnsvizor dashboard web interface](../images/dashboard.png)
+
+- `/querylog` 
+    - **method**: `GET`
+    - **description**: Displays a log of recent DNS queries processed by the resolver. 
+    - **authenticated**: false
```

Comment:
maybe worth to mention that this query log is not something that works at the moment?